scripts/mainmenu.js

```diff
@@ -1,115 +1,147 @@
+		this.panels.model.SetDirectionalLightDirection(0, 1.0, 0.0, 0.0);
+
+		if (GameInterfaceAPI.GetSettingBool('developer')) {
+			$('#ControlsLibraryButton').RemoveClass('hide');
+			$('#PSDumpButton').RemoveClass('hide');
 		}
+
+		this.setMainMenuBackground();
+
+		this.showPlaytestWelcomePopup();
 	}
 
+	/**
+	 * Fired by C++ whenever main menu is switched to.
+	 */
 	static onShowMainMenu() {
-		MainMenuController.onInitFadeUp();
+		this.panels.movie = $('#MainMenuMovie');
+		this.panels.image = $('#MainMenuBackground');
 
-		MainMenuController.videoPanel = $('#MainMenuMovie');
-		MainMenuController.imagePanel = $('#MainMenuBackground');
-
-		MainMenuController.setMainMenuBackground();
+		this.setMainMenuBackground();
 	}
 
+	/**
+	 * Fired by C++ whenever main menu is switched from.
+	 */
 	static onHideMainMenu() {
 		UiToolkitAPI.CloseAllVisiblePopups();
 	}
 
+	/**
+	 * Fired by C++ whenever pause menu (i.e. main menu when in a map) is switched to.
+	 */
 	static onShowPauseMenu() {
-		$.GetContextPanel().AddClass('MainMenuRootPanel--PauseMenuMode');
+		this.panels.cp.AddClass('MainMenuRootPanel--PauseMenuMode');
 	}
 
+	/**
+	 * Fired by C++ whenever pause menu is switched from.
+	 */
 	static onHidePauseMenu() {
-		$.GetContextPanel().RemoveClass('MainMenuRootPanel--PauseMenuMode');
+		this.panels.cp.RemoveClass('MainMenuRootPanel--PauseMenuMode');
 
 		// Save to file whenever the settings page gets closed
 		if (this.activeTab === 'Settings') {
 			$.DispatchEvent('SettingsSave');
 		}
 	}
 
-	static navigateToTab(tab, xmlName, hasBlur = true) {
-		if (tab === 'MapSelection') {
-			$.GetContextPanel()
-				.FindChildTraverse('MainMenuBackgroundMapSelectorImage')
-				.RemoveClass('mapselector__background--hidden');
-		} else {
-			$.GetContextPanel()
-				.FindChildTraverse('MainMenuBackgroundMapSelectorImage')
-				.AddClass('mapselector__background--hidden');
-		}
+	/**
+	 * Switch main menu page
+	 */
+	static navigateToPage(tab, xmlName, hasBlur = true) {
+		this.panels.mapSelectorBackground.SetHasClass('mapselector__background--hidden', tab !== 'MapSelection');
 
-		MainMenuController.contentBlurPanel.visible = hasBlur;
+		this.panels.contentBlur.visible = hasBlur;
 
-		if (MainMenuController.activeTab == tab) {
-			$.DispatchEvent('Activated', $('#HomeButton'), 'mouse');
+		if (this.activeTab === tab) {
+			$.DispatchEvent('Activated', this.panels.homeButton, 'mouse');
 			return;
 		}
 
 		// Check to see if tab to show exists.
 		// If not load the xml file.
-		if (!$.GetContextPanel().FindChildInLayoutFile(tab)) {
-			const newPanel = $.CreatePanel('Panel', this.contentPanel, tab);
+		if (!this.panels.cp.FindChildInLayoutFile(tab)) {
+			const newPanel = $.CreatePanel('Panel', this.panels.pageContent, tab);
 
-			newPanel.Data().elMainMenuRoot = $.GetContextPanel();
-			newPanel.LoadLayout('file://{resources}/layout/' + xmlName + '.xml', false, false);
+			// newPanel.Data().elMainMenuRoot = $.GetContextPanel();
```

Comment:
Commented out code?

Reply:
Oh yeah, think this is from CS:GO and I forgot to remove it. Interesting what they're doing here, passing a parent panel's JSO into a child panel.